core/systems/bluetooth/bluetooth_manager.gd

```diff
@@ -0,0 +1,278 @@
+extends Resource
+class_name BluetoothManager
+
+## Manage and interact with bluetooth devices
+##
+## Allows bluetooth management through bluez
+## Reference: https://wiki.archlinux.org/title/bluetooth#Pairing
+
+var logger := Log.get_logger("BluetoothManager", Log.LEVEL.DEBUG)
+
+var cmd_queue: Array[String] = []
+var current_cmd := ""
+var current_output: Array[String] = []
+var discovered_devices: Array[BluetoothDevice]
+var proc: InteractiveProcess
```

Comment:
Instead of running an interactive process, can you just run these commands using the new `ThreadPool`?

Reply:
That might work. Most of the commands are running in the main thread with immediate execution. The issue so far has been that scan tends to dump its output into the stdout of any process running bluetoothctl, and never returns it output to its own process even with --timeout if done with a SharedThread or InteractiveProcess. Scan is the only process that *needs* asynchronous execution to be useful, the others might benefit just from a performance boost perspective.